hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -83,6 +83,15 @@ protected HiveMetaStoreClient reconnect(HiveMetaStoreClient client) {
     return client;
   }
 
+  @Override
+  protected boolean isConnectionException(Exception e) {
+    if (super.isConnectionException(e) || (e != null && e instanceof MetaException &&
+        e.getMessage().contains("Got exception: org.apache.thrift.transport.TTransportException"))) {
```

Comment:
nit: can we just say:
```
return super.isConnectionException(e) || (e != null && e instanceof MetaException &&
        e.getMessage().contains("Got exception: org.apache.thrift.transport.TTransportException"));
```

Reply:
May lead to poor debugging

Reply:
What do you mean, how does it impact debugging? E.g. not being able to put a breakpoint there? I don't think we should optimize for that in the source code - less verbose is usually better. Plus for debugging purposes, you can always stop at the return line and evaluate the boolean expression in your IDE, or just see what which branch the execution follows after `isConnectionException()`.

Reply:
Thanks @Jecarm for all the fixes.
For me it seems that @marton-bod has a good point. I think this is the last thing and then I would happily push the merge! 😄 Thanks for all your work on this!

Reply:
i fixed it.